plugins/ua_network_pubsub_ethernet.c

```diff
@@ -0,0 +1,422 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ *   Copyright 2018 (c) Kontron Europe GmbH (Author: Rudolf Hoyler)
+ */
+
+#include "ua_plugin_network.h"
+#include "ua_log_stdout.h"
+#include "ua_util.h"
+
+/* **NOTE: Currently ONLY for LINUX** */
+#include <netpacket/packet.h>
+#include <netinet/ether.h>
+
+#include "ua_network_pubsub_ethernet.h"
+
+#define ETHERTYPE_UADP  0xB62C
+
+/* internal structure */
+#define MAX_INTERFACE_NAME 64
+#define MAX_MAC_ADDR       18
+
+/* ethernet network layer specific internal data */
+typedef struct {
+    char interfaceName[MAX_INTERFACE_NAME];
+    char destinationMac[MAX_MAC_ADDR];
+    UA_UInt16 vid;
+    UA_Byte prio;
+    u_int8_t etherSourceAddr[ETH_ALEN];
+} UA_PubSubChannelDataEth;
+
+/* for UDP/TCP exists 'UA_parseEndpointUrl' (ua_util.c).
+ * This is adapted for ethernet here locally.
+ */
+
+static UA_StatusCode
+UA_parseEndpointUrlEthernet(const UA_String *endpointUrl, UA_PubSubChannelDataEth *channelDataEth) {
+    /* related to OPC-UA specification the URL for ethernet should be:
+     *    opc.eth://<host>[:<vid>[.<pcp>]]
+     * with <host> .. MAC address, IP address or name
+     *      <vid>  .. VLAN ID
+     *      <pcp>  .. priority
+     */
```

Comment:
Describe what the <host> looks like. According to OPC-UA Part 14:

"The host is a MAC address, an IP address or a registered name like a hostname. The format 
of a MAC address is six groups of hexadecimal digits, separated by hyphens (e.g. 01-23-45-
67-89-ab). A system may also accept hostnames and/or IP addresses if it provides means to 
resolve it to a MAC address (e.g. DNS and Reverse-ARP)."

I guess resolving any hostname is out of scope for now..